flocker/control/httpapi.py

```diff
@@ -165,12 +178,21 @@ def get_dataset_configuration(self):
         :return: A ``list`` of ``dict`` representing each of dataset
             that is configured to exist anywhere on the cluster.
         """
-        return list(datasets_from_deployment(self.persistence_service.get()))
+        # Perhaps exposing etag should be done in @structured
+        # implementation, if we end up doing it in more than one
+        # place. For now this is simpler:
+        etag = get_configuration_etag(self)
+        return EndpointResponse(
+            OK, list(datasets_from_deployment(self.persistence_service.get())),
+            headers={b"etag": etag})
```

Comment:
`Etag` should be a string surrounded by double quotes (see [RFC](https://tools.ietf.org/html/rfc7232#section-2.3)).